### ISSUE TITLE
Liking and unliking client side pages

### ISSUE THREAD

**Issue by DmitriyVdE** [pnp/pnpjs#2365]
### Category
- [ ] Enhancement
- [ ] Bug
- [x] Question
- [x] Documentation gap/issue

### Version

Please specify what version of the library you are using: [3.5.1]

Please specify what version(s) of SharePoint you are targeting: [Online]

### Expected / Desired Behavior / Question
Hi, I'm currently working on a webpart that has the functionality to like and unlike pages from within it.
This webpart has existed for some time already, but I'm in the process of upgrading the dependencies and re-writing some code.

I've upgraded @pnp/sp from v1 to v3 and started having issues with liking and unliking pages.

I would also like to reference issue #1542 and the supposed fix #1546 since someone already made an issue about this.

I also want to mention that I have tested this on multiple tenants using the old and new versions of the webpart.

EDIT: Followed the documentation at https://pnp.github.io/pnpjs/sp/comments-likes/#like

### Observed Behavior
I retrieve a page, its likes and attempt to like or unlike it using the following code:

```
// weblink is something like '/sites/sitename/SitePages/pagename.aspx'
const item: IItem = await sp.web.getFileByServerRelativePath(webLink).getItem();
const likedByData: ILikeData[] = await item.getLikedBy();
// returns an array of likes and works
await item.like(); // or .unlike()
```

The endpoints in v1 work and are as follows:
```
https://tenantname.sharepoint.com/sites/sitename/_api/Web/Lists(guid(somelistid))/Items(28)/like
https://tenantname.sharepoint.com/sites/sitename/_api/Web/Lists(guid(somelistid))/Items(28)/unlike
```

The endpoints in v3 don't work and are as follows:
```
https://tenantname.sharepoint.com/sites/sitename/_api/Microsoft.Office.Server.ReputationModel.Reputation.SetLike(listID=@a1,itemID=@a2,like=@a3)?@a1=%27{listid}%27&@a2=%2710%27&@a3=true
https://tenantname.sharepoint.com/sites/sitename/_api/Microsoft.Office.Server.ReputationModel.Reputation.SetLike(listID=@a1,itemID=@a2,like=@a3)?@a1=%27{listid}%27&@a2=%2710%27&@a3=false
```
These endpoints return a HTTP 500 Error:
![image](https://user-images.githubusercontent.com/43698501/181298427-2c94b339-fd89-417a-8bd7-e81c0193e178.png)

Now I could just manually do these requests using sphttp, but I just find it weird that the endpoints changed while they used to work.

### Steps to Reproduce
As described above.

**Comment by DmitriyVdE** [comment]
I also found this documentation which also covers liking and unliking pages but it retrieves them in a different way:
https://pnp.github.io/pnpjs/sp/clientside-pages/#like-unlike

```
import { spfi } from "@pnp/sp";
import { IClientsidePage } from "@pnp/sp/clientside-pages";

const sp = spfi(...);

// our page instance
const page: IClientsidePage = await sp.web.loadClientsidePage("/sites/dev/sitepages/home.aspx");

// like this page
await page.like();

// unlike this page
await page.unlike();
```

I tried this but the IClientsidePage does not seem have a function called like() or unlike().

**Comment by juliemturner** [comment]
What you shared above from v1 is liking and item not a page, although with classic it probably worked similarly.

In modern SharePoint, liking for items and pages is different so you need to use the right one with the right thing.

With regard to your second comment, I just tested that code from the documentation and it works fine for me. I note that you left off some key imports that are in the documentation... maybe you did this elsewhere, but that would explain why you say IClientsidePage does not seem to have a function called like/unlike...

```TypeScript
import "@pnp/sp/clientside-pages";
import "@pnp/sp/comments/clientside-page";
```

**Comment by DmitriyVdE** [comment]
I have tried your imports (I only tried `@pnp/sp/clientside-pages` previously) and adding `@pnp/sp/comments/clientside-page` to the imports works. I just find it strange that the import for this would be under sp/comments.

The documentation could also be updated to mention this import being required since it's only mentioned once on the whole page under '[enableComments & disableComments](https://pnp.github.io/pnpjs/sp/clientside-pages/#enablecomments-disablecomments)'.

Anyway, thank you for fast answer!
